### ISSUE TITLE
Two problems about training and decoding

### ISSUE THREAD

**Issue by Marcovaldong** [HawkAaron/E2E-ASR#6]
Hi, @HawkAaron I'm trying to train transducer with pytorch (I prefer to use it rather than MxNet) and I changed the code of this repo following another implementation of MxNet. However, I found the model cannot converge to a good result. Is there something wrong in my code? 

Another problem is that I try to replace the code of [here](https://github.com/HawkAaron/E2E-ASR/blob/master/model.py#L90) with [while loop](https://github.com/HawkAaron/RNN-Transducer/blob/graves2013/model.py#L89). However the model cannot get out of the while loop, is there something different of two implementations?

**Comment by Marcovaldong** [comment]
@HawkAaron I fixed the second problem, that's because of a bug. 

The first one is still existing. Maybe it's also a bug. I'll check my code.

**Comment by HawkAaron** [comment]
@Marcovaldong 
You may compare the results between MxNet and PyTorch, it would be 2% absolute different if not carefully  tuned.

**Comment by Marcovaldong** [comment]
@HawkAaron You means MxNet gets better result, is it? 

Can I compare the loss between MxNet and PyTorch. Now, the Pytorch version cannot converge to the same loss as the MxNet version (the loss of PyTorch version can only converge to 9.1, however the loss of mxnet version can converge to 0.5), and the CER is too high (about 94.5%, the CER of MxNet version is about 1.41%). 

I don't know what problem makes the model doesn't converge.

**Comment by HawkAaron** [comment]
@Marcovaldong Did you solve the problem ? The loss should be comparable, have you reproduce the results on TIMIT ?
